example/cumulus-tf/discover_granules_with_execution_name_prefix_workflow.asl.json

```diff
@@ -0,0 +1,95 @@
+{
+  "Comment": "Discovers new Granules from a given provider",
+  "StartAt": "DiscoverGranules",
+  "TimeoutSeconds": 18000,
+  "States": {
+    "DiscoverGranules": {
+      "Parameters": {
+        "cma": {
+          "event.$": "$",
+          "ReplaceConfig": {
+            "FullMessage": true
+          },
+          "task_config": {
+            "provider": "{$.meta.provider}",
+            "provider_path": "{$.meta.provider_path}",
+            "collection": "{$.meta.collection}",
+            "buckets": "{$.meta.buckets}",
+            "stack": "{$.meta.stack}",
+            "duplicateGranuleHandling": "{$.meta.collection.duplicateHandling}"
+          }
+        }
+      },
+      "Type": "Task",
+      "Resource": "${discover_granules_task_arn}",
+      "Retry": [
+        {
+          "ErrorEquals": [
+            "Lambda.ServiceException",
+            "Lambda.AWSLambdaException",
+            "Lambda.SdkClientException"
+          ],
+          "IntervalSeconds": 2,
+          "MaxAttempts": 6,
+          "BackoffRate": 2
+        }
+      ],
+      "Catch": [
+        {
+          "ErrorEquals": [
+            "States.ALL"
+          ],
+          "ResultPath": "$.exception",
+          "Next": "WorkflowFailed"
+        }
+      ],
+      "Next": "QueueGranules"
+    },
+    "QueueGranules": {
+      "Parameters": {
+        "cma": {
+          "event.$": "$",
+          "ReplaceConfig": {
+            "FullMessage": true
+          },
+          "task_config": {
+            "queueUrl": "${start_sf_queue_url}",
+            "provider": "{$.meta.provider}",
+            "internalBucket": "{$.meta.buckets.internal.name}",
+            "stackName": "{$.meta.stack}",
+            "granuleIngestWorkflow": "${ingest_granule_workflow_name}",
+            "executionNamePrefix": "{$.meta.executionNamePrefix}"
```

Comment:
As I re-look over this, I'm struggling to see what would be setting `meta.executionNamePrefix`? Is this something that is only possible from the integration tests?

Reply:
In the real world, you would define `executionNamePrefix` in the meta of your rule, and that is what would be fed in here. Or, depending on how flexible you need it to be, I could even see a DAAC just hard-coding a value here.

Reply:
Wait, I thought based on https://github.com/nasa/cumulus/pull/1885#discussion_r496870748, that we aren't defining `executionNamePrefix` in the meta of the rule?

On a related note, I'm struggling to see how the `executionNamePrefix` is used by queue-* tasks for a rule that triggers a discover-* workflow